atom/common/promise_util.h

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2018 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef ATOM_COMMON_PROMISE_UTIL_H_
+#define ATOM_COMMON_PROMISE_UTIL_H_
+
+#include <string>
+
+#include "native_mate/converter.h"
+
+namespace atom {
+
+namespace util {
+
+class Promise {
+ public:
+  explicit Promise(v8::Isolate* isolate) {
+    isolate_ = isolate;
```

Comment:
Holding onto an unmanaged pointer that we don't own is inherently risky. It would be better to remove the `isolate_` field.

Perhaps have `Resolve()` and `Reject()` take a context argument instead? That would also let us avoid the deprecate form of those two functions mentioned below

Reply:
Afaik this is how it is done currently all over the native mate codebase.  I'd rather be consistent here

Reply:
Yeap, looking through native_mate that is how we do it there.

I was looking for comparison points in `atom/`, which seems to take isolate arguments when needed, rather than holding onto them in the constructor. I think that's a better way to do it than holding onto a raw pointer, but point taken about us already doing this elsewhere.